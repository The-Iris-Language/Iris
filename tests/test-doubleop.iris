..tests ++ and -- on ints and floats
class Main () {
    public:
        int univ main() {
            int a = 0;
            a++;
            if (a == 1) {
                Olympus.print("successfully incremented a");
            }
            a--;
            if (a == 0) {
                Olympus.print("successfully decremented a"); 
            }

            float f = 4.35;
            f++; 
            if (f == 5.35) {
                Olympus.print("successfully incremented f");
            }
            f--;
            if (f == 4.35) {
                Olympus.print("successfully decremented f");
            }
        }
}