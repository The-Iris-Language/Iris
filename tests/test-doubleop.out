successfully incremented a
successfully decremented a
successfully incremented f
successfully decremented f
